src/StaticWebAssetsSdk/Tasks/DefineStaticWebAssets.Cache.cs

```diff
@@ -0,0 +1,217 @@
+            }
+        }
+
+        internal void TotalUpdate(byte[] propertiesHash, byte[] fingerprintPatternsHash, byte[] propertyOverridesHash, IDictionary<string, ITaskItem> inputsByHash)
+        {
+            _log?.LogMessage(MessageImportance.Low, "Updating cache completely.");
+            GlobalPropertiesHash = propertiesHash;
+            FingerprintPatternsHash = fingerprintPatternsHash;
+            PropertyOverridesHash = propertyOverridesHash;
+            InputHashes = [.. inputsByHash.Keys];
+            _inputHashes = inputsByHash;
+        }
+
+        internal void PartialUpdate(Dictionary<string, ITaskItem> inputHashes)
+        {
+            var newHashes = new HashSet<string>(inputHashes.Keys);
+            var oldHashes = InputHashes;
+
+            if (newHashes.SetEquals(oldHashes))
+            {
+                // If all the input hashes match, then we can reuse all the results.
+                foreach (var cachedAsset in CachedAssets)
+                {
+                    _assets.Add(cachedAsset.Value.ToTaskItem());
+                }
+                foreach (var cachedCopyCandidate in CachedCopyCandidates)
+                {
+                    _copyCandidates.Add(cachedCopyCandidate.Value.ToTaskItem());
+                }
+
+                _cacheUpToDate = true;
+                _log?.LogMessage(MessageImportance.Low, "Cache is fully up to date.");
+                return;
+            }
+
+            var remainingCandidates = new Dictionary<string, ITaskItem>();
+            foreach (var kvp in inputHashes)
+            {
+                var candidate = kvp.Value;
+                var hash = kvp.Key;
+                if (!oldHashes.Contains(hash))
+                {
+                    remainingCandidates.Add(hash, candidate);
+                }
+                else if (CachedAssets.TryGetValue(hash, out var asset))
+                {
+                    _log?.LogMessage(MessageImportance.Low, "Asset {0} is up to date", candidate.ItemSpec);
+                    _assets.Add(asset.ToTaskItem());
+                    if (CachedCopyCandidates.TryGetValue(hash, out var copyCandidate))
+                    {
+                        _copyCandidates.Add(copyCandidate.ToTaskItem());
+                    }
+                }
+            }
+
+            // Remove any assets that are no longer in the input set
+            InputHashes = newHashes;
+            var assetsToRemove = oldHashes.Except(InputHashes);
+            foreach (var hash in assetsToRemove)
+            {
+                CachedAssets.Remove(hash);
+                CachedCopyCandidates.Remove(hash);
+            }
+
+            _inputHashes = remainingCandidates;
+        }
+
+        internal void SetPathAndLogger(string manifestPath, TaskLoggingHelper log) => (_manifestPath, _log) = (manifestPath, log);
+
+        public (IList<ITaskItem> CopyCandidates, IList<ITaskItem> Assets) ComputeOutputs() => (_copyCandidates, _assets);
```

Comment:
Nit: `GetComputedOutputs()` or `GetOutputs()`? Since the actual computing happens elsewhere.